arduino/i2c_slave_code_rccar/i2c_slave_rccar/i2c_slave_rccar.ino

```diff
@@ -101,40 +101,43 @@ bool checkRemote() {
 // to update the motor values
 void receiveEvent(int bytes) {
     // check incoming bytes from master
-    uint8_t read_array[2];
-    int i = 0;
-    for(int i = 0; Wire.available() && i < 2; i++) {
-        read_array[i] = Wire.read();
+    uint8_t read_array[bytes];
+    int read_byte = 0; // number of bytes read
+    for(read_byte = 0; Wire.available() && read_byte < bytes; read_byte++) {
+        read_array[read_byte] = Wire.read();
+    }
+
+    if (read_byte+1==bytes) { // check if we did read all the data
+        movement[0] = read_array[0]; // throttle value
+        movement[1] = read_array[1]; // steering value
+        updateMotors = true;
     }
```

Comment:
Much better! I hadn't noticed that you weren't using ``bytes`` before either.

Reply:
Are you sure it should be ```if (read_byte+1==bytes) ``` ? If the ```Wire.available()``` remains true throughout, the loop ends with  ```read_byte == bytes``` I would have thought?

Reply:
You're totally right. D'oh! We've merged this branch now but I'll push a fix to master.

Reply:
Scratch that. It occurs to me that this code *can't* have been tested, because at the moment, ``movement`` will never be updated. Untested code should *not* be going into the master branch.

@tenxlenx Is this the code you're running on the Arduino at present? If so, the speed and steering values won't be being saved. (If not, then why not...?) I'd appreciate it if you could open a PR with a fix and confirm that this has been tested properly before we push it to master.

Reply:
> Are you sure it should be `if (read_byte+1==bytes) ` ? If the `Wire.available()` remains true throughout, the loop ends with `read_byte == bytes` I would have thought?

You are totally right Thomas read_byte+1 == bytes, I'll fix that asap

Reply:
> Scratch that. It occurs to me that this code _can't_ have been tested, because at the moment, `movement` will never be updated. Untested code should _not_ be going into the master branch.
> 
> @tenxlenx Is this the code you're running on the Arduino at present? If so, the speed and steering values won't be being saved. (If not, then why not...?) I'd appreciate it if you could open a PR with a fix and confirm that this has been tested properly before we push it to master.

@alexdewar I uploaded this to the robot, but this is only to control the robot from the Jetson not saving values to our csv file, so maybe we should move this issue somewhere else,  as it is not directly effecting the recorder program. For now I could just remove the whole receiveEvent() function and re-add it in another pull-request.

Reply:
Ahh ok. In that case feel free to push a fix directly to master for it. It doesn't really need a separate PR.